lib/cretonne/src/verifier.rs

```diff
@@ -614,6 +615,28 @@ impl<'a> Verifier<'a> {
         Ok(())
     }
 
+    fn cfg_integrity(&self, cfg: &ControlFlowGraph) -> Result<()> {
+        for ebb in self.cfg.ebbs() {
+            let cfg_successors = cfg.get_successors(ebb);
+            let cfg_predecessors = cfg.get_predecessors(ebb);
+            for &succ in self.cfg.get_successors(ebb) {
+                if !cfg_successors.contains(&succ) {
```

Comment:
This is correct, but only half way there. This code is checking that `self.cfg.get_successors(ebb)` is a subset of `cfg_successors`. There could be extra elements in cfg_successors` and we wouldn't notice.

Also be careful about introducing accidentally quadratic code. It is theoretically possible for both successor and predecessor sets to get very large, and this algorithm would be quadratic in the size of the sets.

It might be easier to use something like `HashSet::symmetric_difference`. (Or the `BTreeSet` equivalent for nicely ordered EBBs in the error messages).

I expect the CFG implementation to use our own B-tree data structures in the future, but for now it is fine to use the standard library ones in the verifier.